### DISCUSSION TITLE
Exporting component using Listbox fails

### DISCUSSION THREAD

**Discussion by LMinato** [tailwindlabs/headlessui#2839] [Help]
Hi, I am trying to use HeadlessUI as part of an internal react component package. The TS configuration is set to generate `d.ts` files and source maps. But when those settings are active, I am getting a typescript error specifying my exported variable is using a name from @headlessui/react that cannot be named: 
![image](https://github.com/tailwindlabs/headlessui/assets/8792217/00da82d8-8447-4151-9195-c1b99ba5e43a)

The component is declared by using Listbox props like this : 
```
const SelectRoot = forwardRef<
  ElementRef<typeof Listbox>,
  PropsWithChildren<Omit<ComponentPropsWithoutRef<typeof Listbox>, "children">>
>(props, ref) => { /* The component code */ }) 
```

Does someone have had the same issue, or have any idea on how I could fix it ? 
Thanks for your help,

**Comment by deli-ght** [comment]
Did you solve this problem? I encounter same issue!
